src/orion/storage/track.py

```diff
@@ -0,0 +1,705 @@
+        """Insert a new experiment inside the database"""
+        self._get_project(config['name'])
+
+        self.group = self.backend.new_trial_group(
+            TrialGroup(
+                name=experiment_uid(name=config['name'], version=config['version']),
+                project_id=self.project.uid,
+                metadata=to_json(config)
+            )
+        )
+
+        if self.group is None:
+            raise DuplicateKeyError('Experiment was already created')
+
+        config['_id'] = self.group.uid
+        return config
+
+    def update_experiment(self, experiment, where=None, **kwargs):
+        """Update the fields of a given trials
+
+        Parameters
+        ----------
+        experiment: Experiment
+            Experiment object to update
+
+        where: Optional[dict]
+            constraint experiment must respect for the update to take place
+
+        **kwargs: dict
+            a dictionary of fields to update
+
+        Returns
+        -------
+        returns true if the underlying storage was updated
+
+        """
+        raise RuntimeError('You should not update a track experiment')
+
+    def fetch_experiments(self, query, selection=None):
+        """Fetch all experiments that match the query"""
+        new_query = {}
+        for k, v in query.items():
+            if k == 'name':
+                new_query['metadata.name'] = v
+            elif k.startswith('metadata'):
+                new_query['metadata.{}'.format(k)] = v
+            else:
+                new_query[k] = v
+
+        groups = self.backend.fetch_groups(new_query)
+
+        experiments = []
+        for group in groups:
+            version = group.metadata.get('version', 0)
+            metadata = group.metadata.get('metadata', {})
+            experiments.append({
+                '_id': group.uid,
+                'name': group.project_id,
+                'metadata': metadata,
+                'version': version
+            })
+
+        return experiments
+
+    def register_trial(self, trial):
+        """Create a new trial to be executed"""
+        stamp = datetime.datetime.utcnow()
+        trial.submit_time = stamp
+
+        metadata = dict()
+        metadata['params_types'] = {remove_leading_slash(p.name): p.type for p in trial.params}
+        metadata['submit_time'] = to_json(trial.submit_time)
+        metadata['end_time'] = to_json(trial.end_time)
+        metadata['worker'] = trial.worker
+        metadata['metric_types'] = {remove_leading_slash(p.name): p.type for p in trial.results}
+        metadata['metric_types'][self.objective] = 'objective'
+        heartbeat = to_json(trial.heartbeat)
+        if heartbeat is None:
+            heartbeat = 0
+        metadata['heartbeat'] = heartbeat
+
+        metrics = defaultdict(list)
+        for p in trial.results:
+            metrics[p.name] = [p.value]
+
+        self.current_trial = self.backend.new_trial(TrackTrial(
```

Comment:
When does `new_trial` return None? It's hard to guess from [here](https://github.com/Delaunay/track/blob/d8ffb0024bc0ee8720f6889c31c92aa989e361d7/track/client.py#L224)